sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -900,26 +900,31 @@ public virtual Response UpdateMetricAnomalyDetectionConfiguration(string detecti
         /// </summary>
         /// <param name="metricId">Filters the result. The unique identifier of the metric to which the returned <see cref="AnomalyDetectionConfiguration"/>s apply.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <returns>A collection of <see cref="AnomalyDetectionConfiguration"/> items.</returns>
+        /// <returns>An <see cref="AsyncPageable{T}"/> containing the collection of <see cref="AnomalyDetectionConfiguration"/>s.</returns>
         /// <exception cref="ArgumentNullException"><paramref name="metricId"/> is null.</exception>
         /// <exception cref="ArgumentException"><paramref name="metricId"/> is empty or not a valid GUID.</exception>
-        public virtual async Task<Response<IReadOnlyList<AnomalyDetectionConfiguration>>> GetMetricAnomalyDetectionConfigurationsAsync(string metricId, CancellationToken cancellationToken = default)
+        public virtual AsyncPageable<AnomalyDetectionConfiguration> GetMetricAnomalyDetectionConfigurationsAsync(string metricId, CancellationToken cancellationToken = default)
         {
             Guid metricGuid = ClientCommon.ValidateGuid(metricId, nameof(metricId));
 
-            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorAdministrationClient)}.{nameof(GetMetricAnomalyDetectionConfigurations)}");
-            scope.Start();
-
-            try
-            {
-                Response<AnomalyDetectionConfigurationList> response = await _serviceRestClient.GetAnomalyDetectionConfigurationsByMetricAsync(metricGuid, cancellationToken).ConfigureAwait(false);
-                return Response.FromValue(response.Value.Value, response.GetRawResponse());
-            }
-            catch (Exception e)
+            async Task<Page<AnomalyDetectionConfiguration>> FirstPageFunc(int? pageSizeHint)
```

Comment:
This is for me to make sure I understand. So [there isn't](https://github.com/Azure/azure-rest-api-specs/blob/3cbc984fcf0fab278b9c28175319f65db1b9162a/specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json#L2697) a nextfunction because this is a collection that you are masking in a pageable?

Reply:
Correct. We get the whole collection on the first service call, which is made by `FirstPageFunc`. In a real pageable, subsequent calls would be made by `NextPageFunc` to retrieve the following pages, but they are not necessary/supported here.

Reply:
Is this for future proof in case the service turns it into a Pageable? or why?

Reply:
I would say this and consistency are the main reason, but that's my personal opinion. [Our guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-paging) are explicit about returning pageables in methods that return collections.

Reply:
Thank you for the link! I forgot about that.
I wonder why we didn't do this in TA though :|